syntax = "proto2";

import "workloaddriver.proto";

message AckProto {};

message ExperimentParams {
    required int32 qps_sample_rate = 2 [default = 10];
    required int32 max_qps_second = 3 [default = -1];
    required int32 runtime = 4 [default = 10];
    required bool unlimited_stream = 5 [default = false];
    required int32 op_count = 6 [default = 10000];
    required int32 contains = 7 [default = 80];
    required int32 insert = 8 [default = 10];
    required int32 remove = 9 [default = 10];
    required int32 key_lb = 10 [default = 0];
    required int32 key_ub = 11 [default = 1000000];
    required int32 region_size = 12 [default = 22];
    required int32 thread_count = 13 [default = 1];
    required int32 node_count = 14 [default = 0];
    required int32 qp_max = 15 [default = 30];
    required int32 node_id = 16 [default = -1];
}

message ResultProto {
    optional ExperimentParams params = 1;
    repeated rome.WorkloadDriverProto driver = 2; 
}